apps/extension/src/ui/atoms/balances.ts

```diff
@@ -46,12 +47,27 @@ const rawBalancesAtom = atom((get) => {
 })
 
 const filteredRawBalancesAtom = atom(async (get) => {
-  const [tokens, balances] = await Promise.all([
+  const [tokens, chains, accounts, balances] = await Promise.all([
     get(activeTokensWithTestnetsMapAtom),
+    get(activeChainsWithTestnetsMapAtom),
+    get(accountsMapAtom),
     get(rawBalancesAtom),
   ])
 
-  return balances.filter((b) => tokens[b.tokenId])
+  // exclude invalid balances
+  return balances.filter((b) => {
```

Comment:
I don't understand why there was a need for a change to logic here? Isn't this filtering already done with our existing logic for ledger accounts?

Reply:
> because we might have an account which is still the same account but is now locked to a different chain?

yes that, and also i noticed that i saw tokens that shouldn't be there, 
ie : polkadot's DOT where appearing on the portfolio of a polkadot asset hub account

Feels pretty nice to have this filtering at this low level, making it unnecessary to think about it everywhere else in the UI